x/vote/keeper/msg_server.go

```diff
@@ -0,0 +1,83 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	vote "github.com/axelarnetwork/axelar-core/x/vote/exported"
+	"github.com/axelarnetwork/axelar-core/x/vote/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns a new msg server instance
+func NewMsgServerImpl(keeper Keeper) types.MsgServiceServer {
+	return msgServer{
+		Keeper: keeper,
+	}
+}
+
+// Vote handles vote request
+func (s msgServer) Vote(c context.Context, req *types.VoteRequest) (*types.VoteResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	voter := s.snapshotter.GetOperator(ctx, req.Sender)
+	if voter == nil {
+		return nil, fmt.Errorf("account %v is not registered as a validator proxy", req.Sender.String())
+	}
+
+	poll := s.GetPoll(ctx, req.PollKey)
+
+	switch {
+	case poll.Is(vote.Expired):
+		return &types.VoteResponse{Log: fmt.Sprintf("vote for poll %s already %s", req.PollKey, vote.Expired.String())}, nil
```

Comment:
is this new? I thought in the past we still allowed voting on expired polls. I'm not opposed to this though

Reply:
It has been there for a while. I think we allow override for an expired vote.